### PR TITLE
feat: add snippets for nextjs

### PR DESCRIPTION
**Pull request by karyanayandi**

### CODE REVIEW

File: snippets/javascript/next-ts.json [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  "Use sized Next Image component": {
+    "prefix": "usim",
+    "body": "<Image alt=\"$1\" src=\"$1\" width=\"$1\" height=\"$1\" />",
+    "description": "Use sized Next Image component"
+  },
+  "Use unsized Next Image component": {
+    "prefix": "usimu",
+    "body": "<Image alt=\"$1\" src=\"$1\" unsized=\"$1\" />",
+    "description": "Use sized Next Image component"
+  }
```
Comment (OkelleyDevelopment) [inline snippets/javascript/next-ts.json:171]: You can remove these lines as they are already present in the `next.json` file.

**Review by OkelleyDevelopment — CHANGES_REQUESTED** [review]
**General Feedback**
These look like good ones to add and the filenames are consistent. :sunglasses: 

**Critiques / Requested Changes**

The reason I marked lines 95 - 171 in `next-ts.json` for removal is because they already exist in the your other file `next.json`. 

If a user wished to have them for the TypeScript development, then they could extend the file type `typescriptreact` to include the `javascriptreact` snippets since all JS is valid TS. 

As an example:
```lua
require("luasnip").filetype_extend("typescriptreact", { "javascriptreact" })
```

In the end, this helps us maintain a slimmer snippet base. If you have any questions/concerns feel free to drop them here and we can discuss further :sunglasses:

File: snippets/javascript/next-ts.json (hunk 1 above)
Comment (karyanayandi) [reply snippets/javascript/next-ts.json:171]: okay. i understand

Comment (OkelleyDevelopment) [reply snippets/javascript/next-ts.json]: Thanks for the quick changes :+1:

**Review by OkelleyDevelopment — APPROVED** [review]
**General Feedback** 

Still looking good *and* will stay maintainable/ extendable for the future!! :sunglasses:

**Pull request merged**